### PR TITLE
Improve the behaviour of `shlex.split` on Windows

### LINKED ISSUE #781: Pip args are incorrectly parsed on Windows
**Describe the bug**

On Windows systems, some characters are lost when pip args are parsed with ``shlex.split()`` (at least backslashes), for example, when setting an option with a path. This is especially important when trying to use a directory with ``--find-links``.
https://github.com/pypa/pipx/blob/51fd28e4b33432d8fead817781ea5ae001519127/src/pipx/main.py#L134

**How to reproduce**
```
pipx install --pip-args="--find-links D:\MODULES_WHEEL\SDK" --verbose internal-package
```

```
pipx >(setup:717): pipx version is 0.16.4
pipx >(setup:718): Default python interpreter is 'c:\\users\\USERNAME\\appdata\\local\\programs\\python\\python36\\python.exe'
pipx >(package_name_from_spec:323): Determined package name: internal-package
pipx >(package_name_from_spec:324): Package name determined in 0.0s
pipx >(run_subprocess:135): running c:\users\USERNAME\appdata\local\programs\python\python36\python.exe -m venv --without-pip C:\Users\USERNAME\.local\pipx\venvs\internal-package
pipx >(run_subprocess:135): running C:\Users\USERNAME\.local\pipx\venvs\internal-package\Scripts\python.exe -c import sysconfig; print(sysconfig.get_path('purelib'))
pipx >(run_subprocess:135): running C:\Users\USERNAME\.local\pipx\shared\Scripts\python.exe -c import sysconfig; print(sysconfig.get_path('purelib'))
pipx >(run_subprocess:135): running C:\Users\USERNAME\.local\pipx\venvs\internal-package\Scripts\python.exe --version
pipx >(_parsed_package_to_package_or_url:128): cleaned package spec: internal-package
pipx >(run_subprocess:135): running C:\Users\USERNAME\.local\pipx\venvs\internal-package\Scripts\python.exe -m pip install --find-links D:MODULES_WHEELSDK internal-package
pipx >(subprocess_post_check_handle_pip_error:297): 'C:\\Users\\USERNAME\\.local\\pipx\\venvs\\internal-package\\Scripts\\python.exe -m pip install --find-links D:MODULES_WHEELSDK internal-package' failed
pipx >(subprocess_post_check_handle_pip_error:315): Fatal error from pip prevented installation. Full pip output in file:
    C:\Users\USERNAME\.local\pipx\logs\cmd_2021-12-16_17.05.01_pip_errors.log

Some possibly relevant errors from pip install:
    ERROR: Could not find a version that satisfies the requirement internal-package (from versions: none)
    ERROR: No matching distribution found for internal-package
pipx >(rmdir:43): removing directory C:\Users\USERNAME\.local\pipx\venvs\internal-package
Error installing internal-package.
```

**Expected behavior**

The paths provided in pip args should not be altered.

### PR DESCRIPTION
**Pull request by meowmeowmeowcat**
* [x] I have added an entry to `docs/changelog.md`

## Summary of changes

Fixes #781 

On Windows, `shlex.split` will make path like `D:\MODULES_WHEEL\SDK` become `D:MODULES_WHEELSDK`, which may cause issues when running `--pip-args`.
This PR adds `posix` argument in `shlex.split` so that it can handle paths on Windows properly.
## Test plan

On Windows, 
```py
print(shlex.split("D:\MODULES_WHEEL\SDK arg1 arg2"))
```
Output: 
```
['D:MODULES_WHEELSDK', 'arg1', 'arg2']
```

With `posix` argument:
```py
print(shlex.split("D:\MODULES_WHEEL\SDK arg1 arg2", posix="win" not in sys.platform))
```
Output:
```
['D:\\MODULES_WHEEL\\SDK', 'arg1', 'arg2']
```

### CODE REVIEW

**Comment by uranusjr** [comment]
Is it possible to add a test for this? The logic looks good to me; `posix=False` is not technically compatible with Windows (either cmd.exe or Powershell), but at least produces a better result. (Maybe say “improve” instead of “fix”?)

**Comment by meowmeowmeowcat** [comment]
@uranusjr I've added a test. Could you approve running the workflows? Thanks!

File: src/pipx/main.py [hunk 1]
```diff
@@ -132,7 +132,9 @@ def get_pip_args(parsed_args: Dict[str, str]) -> List[str]:
         pip_args += ["--index-url", parsed_args["index_url"]]
 
     if parsed_args.get("pip_args"):
-        pip_args += shlex.split(parsed_args.get("pip_args", ""))
+        pip_args += shlex.split(
+            parsed_args.get("pip_args", ""), posix="win" not in sys.platform
```
Comment (gaborbernat) [inline src/pipx/main.py:136]: wouldn't be more appropriate for this to be `"win32" != sys.platform`

Comment (meowmeowmeowcat) [reply src/pipx/main.py:136]: @gaborbernat I've tested and using `"win32" != sys.platform` will still output `['D:MODULES_WHEELSDK', 'arg1', 'arg2']`

File: docs/changelog.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 dev
 
 - [docs] Fix the command for [installing development version](https://pypa.github.io/pipx/installation/#install-pipx-development-versions). (#801)
+- Improve the behaviour of `shlex.split` on Windows (#794)
```
Comment (cs01) [inline docs/changelog.md:4]: This is technically correct, but users won't know what to make of it. Could you mention what this fixes from an end user perspective?

**Comment by cs01** [comment]
Looks like a couple of the tests are failing.

**Review by cs01 — CHANGES_REQUESTED** [review]
Looks like a good improvement, but CI tests need to be fixed first 

edit: I see CI was already broken on the main branch. I'm working on fixing it now.

**Review by cs01 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**